proposals/0387-bls-pubkey-management-in-vote-account.md

```diff
@@ -0,0 +1,222 @@
+---
+simd: '0387'
+title: BLS Pubkey management in vote account
+authors:
+  - Sam Kim (Anza)
+  - Quentin Kniep (Anza)
+  - Wen Xu (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-10-27
+feature: (fill in with feature key and github tracking issues once accepted)
+---
+
+## Summary
+
+This proposal specifies in detail how a BLS public key can be generated by
+users via updated existing tools and how they can put the generated BLS public
+keys into their vote accounts for voting in Alpenglow.
+
+It also describes in detail the data structure changes needed.
+
+## Motivation
+
+The Alpenglow SIMD (326) described the new consensus protocol which will be
+launched on Solana. The protocol requires efficient and safe aggregation of
+validator votes to succinctly prove certain state transitions can safely happen
+(for example, 60% of the validators voted to skip a slot). The ed25519
+signatures we currently use are not the best fit for this purpose, so instead
+we will be using the Boneh–Lynn–Shacham (BLS) aggregate signature scheme to
+sign Alpenglow votes.
+
+However, the BLS public key is entirely different from an ed25519 public key
+(as BLS operates over a different elliptic curve), so we can’t naively reuse
+the current ed25519 public keys in vote accounts either. We have to add a BLS
+public key into each vote account and make that a requirement a little before
+Alpenglow launches.
```

Comment:
wordsmith: "Each validator must add a BLS public into their vote account before the network enables Alpenglow in order to vote"